src/Symfony/Component/Serializer/Tests/SerializerTest.php

```diff
@@ -633,6 +638,56 @@ public function testDeserializeAndUnwrap()
             $serializer->deserialize($jsonData, __NAMESPACE__.'\Model', 'json', [UnwrappingDenormalizer::UNWRAP_PATH => '[baz][inner]'])
         );
     }
+
+    public function testCollectDenormalizationErrors()
```

Comment:
Maybe use a data provider to able to add new cases like for variadic argument

Reply:
The data provider is suitable for cases where the assertion code is always the same for different data. Perhaps it would be better to write a test for each case here.

Reply:
Either way, I just think this one is getting a bit hard to read.
Maybe add a new test for the `UnwrappingDenormlizer` because I think you don't handle it already and the key should be added to error to match the input paylod.

Reply:
Can you provide a test case for `UnwrappingDenormlizer`?

Reply:
There is already a few tests that uses it.
The idea is to wrap the data into container: `{"container:" {"foo": "string", ...}}`
And check that the property path for each errors starts with `container.`